### ISSUE TITLE
Revert the <call-expression> shortcut for "put"

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#2997]
When `put` was made optional in #2967, a shorthand that came along with this is the ability to leave off the `fieldname :=` as long as the right-hand-side expression is the name of a function, since this could create a field of the same name as the function. E.g.:

```
$ zq -version
Version: v0.29.0-488-gd30daf4a

$ echo '{"foo": "BAR"}' | zq 'to_lower(foo)' -
{foo:"BAR",to_lower:"bar"}
```

When I wrote the related docs updates in #2986, I found it a little awkward to include this shortcut in the summary of the new syntax. The best idea we had was one from @mattnibs to describe the function example as a `<call-expression>`:

```
[put] (<field> := <expression>) | <call-expression> [, (<field> := <expression>) | <call-expression> ...]
```

The docs and the code are now in sync, but this has given us reason to pause as a team and reflect on whether we like this as users. The current consensus is now that it feels _weird_.

We originally rationalized doing it on the grounds that aggregations behave like this, e.g., a Zed query consisting of just `count()` creates a field called `count` to hold the result. I've claimed this feels a little apples & oranges: When doing an aggregation, you're effectively replacing your original data (which might have dozens of fields) with new summarized data (which might be as narrow as a single field holding a single value, as in the `count()` example). Meanwhile, adding a new field called `to_lower` to what's potentially a wide sea of existing fields seems strange. I expect most users would be inclined to use `:=` to assign a more meaningful field name in that case. And if it's true that users may rarely invoke this shortcut, it calls into question what we're getting for the combined cost of:

1.  The parser being slightly more complicated & expensive to support
2.  The docs being more complicated to write/extend
3.  The docs being more complicated for the user to read/retain

In conclusion, we've reached consensus on backing out this shortcut.

**Comment by mattnibs** [comment]
Just wanted to clarify here that "revert" isn't the right word for the title of this ticket. This will also change functionality existing prior to my implied put pr since a put proc (and other operations) will require explicit assignments and get rid of implied assignments on function calls.

**Comment by philrz** [comment]
Verified in Zed commit 7db68a5.

Before the change in the linked PR, the following would have put the function's result in a newly created field called `to_lower`, but now we see a helpful error message.

```
$ zq -version
Version: v0.29.0-513-g7db68a51

$ echo '{"greeting": "HELLO"}' | zq -z 'to_lower(greeting)' -
bad expression in filter: function "to_lower" does not return a boolean value
```

An explicit field name and `:=` for assignment is now required.

```
$ echo '{"greeting": "HELLO"}' | zq -z 'quieter:=to_lower(greeting)' -
{greeting:"HELLO",quieter:"hello"}
```

Also, just to call attention to what else the error message is saying, non-assignment calls to functions that return booleans effectively become filters, e.g.,

```
$ echo '{"greeting": "HELLO"}' | zq -z 'has(greeting)' -
{greeting:"HELLO"}

$ echo '{"greeting": "HELLO"}' | zq -z 'has(something_else)' -
[no output]
```

[`put` docs](https://github.com/brimdata/zed/tree/main/docs/language/operators#put) have also been updated.

Thanks @mattnibs!
